test/org/freedesktop/gstreamer/SampleTest.java

```diff
@@ -46,15 +47,42 @@ public void testGetCaps() {
     		Caps caps = sample.getCaps();
     		Structure struct = caps.getStructure(0);
     		String name = struct.getName();
-    		assertEquals(name, "video/x-raw");
+    		assertEquals("video/x-raw", name);
     	});
     }
 
     @Test
     public void testGetBuffer() {
     	SampleTester.test((Sample sample) -> {
     		Buffer buffer = sample.getBuffer();
-    		assertEquals(buffer.getMemoryCount(), 1);
+    		assertEquals(1, buffer.getMemoryCount());
     	});
     }
+    
+    @Test
+    public void testSetBuffer() {
+    	// since gst 1.16, sample is recycled and keep a reference on the last buffer received  
+    	SampleTester.test((Sample sample) -> {
+    		Buffer buffer = sample.getBuffer();
+
+    		int refCount = buffer.getRefCount();
+    		    		
+    		if (Gst.getVersion().checkSatisfies(new Version(1, 16))) {
```

Comment:
This is not how to test features requiring above 1.8.

Firstly, make sure to call init() with `Gst.init(Gst.getVersion(), ...testName...);` as at https://github.com/gstreamer-java/gst1-java-core/blob/master/test/org/freedesktop/gstreamer/PromiseTest.java#L44

Secondly call `TestAssumptions.requireGstVersion(1, 16);` as first line of test - as at https://github.com/gstreamer-java/gst1-java-core/blob/master/test/org/freedesktop/gstreamer/PromiseTest.java#L54